code/modules/urist/modules/shipbattles/weapons.dm

```diff
@@ -282,6 +282,112 @@
+	switch(safety)
+		if(0) to_chat(user, "<span class='warning'>The safeties have been disabled.</span>")
+		if(1) to_chat(user, "<span class='notice'>The safeties are enabled.</span>")
+	if(is_rigged)
+		to_chat(user, "<span class='warning'>There is a [attached_device] attached to the warhaed.</span>")
+
+/obj/item/shipweapons/torpedo_warhead/attackby(var/obj/item/I, mob/user as mob)
+	if(istype(I, /obj/item/weapon/crowbar))
+		if(riggedstate == 1 && !attached_device) // can't close it if it's got something it's not supposed to have.
+			to_chat(user, "<span class='notice'>You carefully close the warhead's circuitry panel.</span>")
+			riggedstate = 0
+			playsound(src.loc, 'sound/items/Crowbar.ogg', 50, 1)
+		else if(!riggedstate)
+			to_chat(user, "<span class='notice'>You carefully lever open the warhead's circuitry panel.</span>")
+			riggedstate++
+			playsound(src.loc, 'sound/items/Crowbar.ogg', 50, 1)
+		else
+			to_chat(user, "<span class='notice'>You can't close the panel. Remove the [attached_device] first.</span>")
+
+	else if(istype(I, /obj/item/weapon/wirecutters))
+		if(riggedstate == 3)
+			to_chat(user, "<span class='notice'>You snip the wire attached to the warhead's detonation circuit.</span>")
+			riggedstate = 2
+			playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+		else if(riggedstate == 2)
+			to_chat(user, "<span class='notice'>You carefully undo the modifications to the warhead's circuitry.</span>")
+			riggedstate = 1
+			playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+		else if(riggedstate == 1)
+			to_chat(user, "<span class='notice'>You begin to carefully modify the circuitry of the warhead.</span>")
+			if(do_after(user,40))
+				to_chat(user, "<span class='notice'>You have modified the torpedo warhead's internal circuitry. It can now be wired up and attached to something.</span>")
+				riggedstate++
+				playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+		else if(is_rigged && attached_device)
+			to_chat(user, "<span class='notice'>You carefully begin to remove [attached_device] from the warhead's internals.</span>")
+			if(do_after(user,40))
+				to_chat(user, "<span class='notice'>You carefully remove [attached_device] from the warhead's internals.</span>")
+				var/obj/item/device/assembly/A = attached_device
+				A.forceMove(get_turf(user))
+				A.holder = null
+				attached_device = null
+				is_rigged = 0
+				playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+		return
+	else if(istype(I, /obj/item/device/multitool))
+		if(!riggedstate)
+			return
+		if(!safety)
+			to_chat(user, "<span class='notice'>You begin to re-engage the built-in safeties.</span>")
+			if(do_after(user,40))
+				to_chat(user, "<span class='notice'>The warhead beeps softly, indicating you have re-enabled it's safeties!</span>")
+				safety = 1
+				playsound(src.loc, 'sound/machines/buttonbeep.ogg', 25, 0, 10)
+		if(safety)
```

Comment:
All of this can become a single toggle with the following changes.
`safety = !safety`
and 
`to_chat(user, "<span class='notice'>You begin to [safety ? "disable" : "re-engage"] the built-in safeties.</span>")`

Reply:
Not sure how to swing this and modify the icon state at the same time.

Reply:
Icon_state is just a string, same as the rest.
`icon_state = "torpedowarhead-open-mod[safety ? "" : "-armed"]"`

Reply:
In the next commit.